src/MerchantCenter/PriceBenchmarks.php

```diff
@@ -101,15 +310,145 @@ public function get_summary(): array {
+		$benchmark_counts = $this->get_price_benchmark_counts_data( $benchmark_counts_result );
+
+		return [
+			'total_products' => $benchmark_counts['total'] ?? 0, // Total products
+			'price_unknown'  => $benchmark_counts[0] ?? 0, // Unknown/missing
+			'price_lower'    => $benchmark_counts[1] ?? 0, // Lower price
+			'price_similar'  => $benchmark_counts[2] ?? 0, // Similar price
+			'price_higher'   => $benchmark_counts[3] ?? 0, // Higher price
+		];
+	}
+
+	/**
+	 * Converts raw benchmark counts from the database to an associative array.
+	 *
+	 * @param array $rows Raw benchmark counts result from the database.
+	 * @return array Associative array with counts for each price comparison group and total.
+	 */
+	public function get_price_benchmark_counts_data( array $rows ): array {
+		// Convert the results to a more usable format
+		$counts = [];
+		$total  = 0;
+		foreach ( $rows as $row ) {
+			$price_compared_value            = (int) $row['price_compared_with_benchmark'];
+			$counts[ $price_compared_value ] = (int) $row['count'];
+			$total                          += $counts[ $price_compared_value ];
+		}
+
+		// Make sure all possible values are represented (0, 1, 2, 3)
+		$all_values = [ 0, 1, 2, 3 ];
+		foreach ( $all_values as $value ) {
+			if ( ! isset( $counts[ $value ] ) ) {
+				$counts[ $value ] = 0;
+			}
+		}
+
+		$counts['total'] = $total;
+
+		return $counts;
+	}
+
+	/**
+	 * Retrieves formatted price benchmarks data from the local database.
+	 *
+	 * @param array $args {
+	 *     Optional. Arguments to filter and paginate results.
+	 *
+	 *     @type array|null $include  List of product IDs to include. Default null.
+	 *     @type int        $offset   Offset for the results. Default 0.
+	 *     @type int        $limit    Maximum number of items returned. Default 10.
+	 *     @type string     $search   Search string to filter results. Default null.
+	 *     @type string     $order    Sort order: 'ASC' or 'DESC'. Default 'DESC'.
+	 *     @type string     $orderby  Attribute to sort by. Default 'id'.
+	 * }
+	 * @return array {
+	 *     @type array $results List of formatted price benchmarks.
+	 *     @type int   $total   Total number of price benchmarks available.
+	 * }
+	 */
+	public function get_price_benchmarks_data( array $args = [] ): array {
+		$defaults = [
+			'include' => null,
+			'offset'  => 0,
+			'limit'   => 10,
+			'search'  => null,
+			'order'   => 'DESC',
+			'orderby' => 'mc_insights_effectiveness',
+		];
+
+		$args = array_merge( $defaults, $args );
+
+		/** @var MerchantPriceBenchmarksQuery $query */
+		$query = $this->container->get( MerchantPriceBenchmarksQuery::class );
+
+		// Apply filters.
+		if ( ! empty( $args['include'] ) && is_array( $args['include'] ) ) {
+			$query->where( 'product_id', $args['include'], 'IN' );
+		}
+
+		if ( ! empty( $args['search'] ) ) {
+			$search = trim( $args['search'] );
+			// TODO: Implement a search mechanism.
```

Comment:
We could consider adding product title to the price benchmarks DB if it's easier, or search the products to get a list of IDs and then use those IDs like `$query->where( 'product_id', $results, 'IN' )`

Reply:
If we store the title in the database and someone later changes the product title in the WC store, will the changes sync quickly? If not, we might run into issues with the DB approach. What do you think?